Source/ASImageNode+AnimatedImage.mm

```diff
@@ -79,6 +80,11 @@ - (void)_locked_setAnimatedImage:(id <ASAnimatedImageProtocol>)animatedImage
       [self setCoverImage:nil];
   }
   
+  // Animated image can take while to dealloc, do it off the main queue
+  if (previousAnimatedImage != nil) {
+    ASPerformBackgroundDeallocation(&previousAnimatedImage);
+  }
+  
   [self animatedImageSet:_animatedImage previousAnimatedImage:previousAnimatedImage];
```

Comment:
`previousAnimatedImage` will be nilled out by this function. Call this after calling `-animatedImageSet:previousAnimatedImage:`

Reply:
Oooh, good catch!